### ISSUE TITLE
Error: variableGrads() expects at least one of the input variables to be trainable in optimizer.computeGradients()

### ISSUE THREAD

**Issue by jaxball** [tensorflow/tfjs#153]
If you would like to get help from the community, our [Google group](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) might be a better fit.

#### TensorFlow.js version
0.7.0
#### Browser version
Chrome Version 65.0.3325.181 (Official Build) (64-bit)

#### Describe the problem or feature request
I've previously developed a _deeplearn.js_ project using MobileNet for inference. Since rebranding to _tensorflow.js_ however, we were able to obtain the gradients with respect to all layer variables and modify the original input image using the `mnist-core` (new in tfjs) example for custom visualization. When we tried to port that to my existing _deeplearn.js_ project however, we get the following error:    
```
Error: variableGrads() expects at least one of the input variables to be trainable, but none of the 0 variables is trainable.

get_adv_xs
C:/Users/jasonlin/Desktop/acv/advis/src/model.js:98
   95 |    // let loss_grad_func = dl.grad( () => { return _loss });
   96 |    // let grad = loss_grad_func(x_2d);
   97 | 
>  98 |    const cost = optimizer.computeGradients(() => {
   99 |      adv_img = generate_adv_xs(hardlabels, x_2d, resAll.logits, 0.2);
  100 |      return dl.losses.softmaxCrossEntropy(hardlabels, resAll.logits).mean();
  101 |    });
```

We suspect that with the way the new tf.loadModel() works, the model is defined locally (correct me if I'm wrong, fetched from [](https://storage.googleapis.com/tfjs-models/tfjs/mobilenet_v1_0.25_224/model.json))). So I tried looking for the equivalent in _deeplearn.js_' MobileNet definition. In `deeplearn/models/mobilenet.ts load()`, I found an argument passed to `dl.CheckpointLoader()`: [](https://storage.googleapis.com/learnjs-data/checkpoint_zoo/mobilenet_v1_1.0_224/). However unlike in the tfjs example where I'm able to download this model definition via my browser, when I tried to download this model I appear to not have access. 

I understand `CheckpointLoader` is deprecated, but we would just like to obtain the gradient of loss function with respect to the input layer and use that to alter the input image. Is there anyway I could get around this?

#### Code to reproduce the bug / link to feature request
```
const cost = optimizer.computeGradients(() => {
      adv_img = generate_adv_xs(hardlabels, x_2d, resAll.logits, 0.2);
      return dl.losses.softmaxCrossEntropy(hardlabels, resAll.logits).mean();
});

function generate_adv_xs(label, x_2d, logits, eps) {
    let _loss = dl.losses.softmaxCrossEntropy(label, logits);
    let loss_grad_func = dl.grad( () => { return _loss });
    let grad = loss_grad_func(x_2d);
    let scaled_grad = scale_grad(grad, eps);
    let perturbed_img = dl.add(x_2d, scaled_grad)
      .clipByValue(MIN_PIXEL_VALUE, MAX_PIXEL_VALUE);
    return perturbed_img;
}
```

**Comment by tafsiri** [comment]
@jaxball Thanks for reporting this, this should work in theory but a couple of people have taken a look and have found a few bugs which we will be working on. In the meantime the quickest way to work around this maybe to use the old checkpoint loading style in your project until this is resolved. 

We moved the CheckpointLoader to this repo https://github.com/PAIR-code/deeplearnjs-legacy-loader, you could just copy the needed code (or copy it in from your deeplearn project) to load mobilenet.

**Comment by nsthorat** [comment]
We don't have gradients defined for depthwise convs, so MobileNet gradients should have worked in the old way. Do you mean SqueezeNet? If so, I would just use the legacy loader for now. We're working on fixing gradients for `model.predict()` right now, which is causing some errors.
